basic_nodes/src/main/java/club/mondaylunch/gatos/basicnodes/ListContainsNodeType.java

```diff
@@ -0,0 +1,45 @@
+package club.mondaylunch.gatos.basicnodes;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+import club.mondaylunch.gatos.core.data.DataBox;
+import club.mondaylunch.gatos.core.data.DataType;
+import club.mondaylunch.gatos.core.data.ListDataType;
+import club.mondaylunch.gatos.core.graph.connector.NodeConnector;
+import club.mondaylunch.gatos.core.graph.type.NodeType;
+
+public class ListContainsNodeType extends NodeType.Process {
+    @Override
+    public Map<String, DataBox<?>> settings() {
+        return Map.of();
+    }
+
+    @Override
+    public Set<NodeConnector.Input<?>> inputs(UUID nodeId, Map<String, DataBox<?>> settings, Map<String, DataType<?>> inputTypes) {
+        return Set.of(
+            new NodeConnector.Input<>(nodeId, "list", ListDataType.GENERIC_LIST),
+            new NodeConnector.Input<>(nodeId, "element", DataType.ANY));
+    }
+
+    @Override
+    public Set<NodeConnector.Output<?>> outputs(UUID nodeId, Map<String, DataBox<?>> settings, Map<String, DataType<?>> inputTypes) {
+        return Set.of(
+            new NodeConnector.Output<>(nodeId, "output", DataType.BOOLEAN));
+    }
+
+    @Override
+    public Map<String, CompletableFuture<DataBox<?>>> compute(Map<String, DataBox<?>> inputs, Map<String, DataBox<?>> settings, Map<String, DataType<?>> inputTypes) {
+        var inputListType = inputTypes.get("list");
+        var elementType = inputTypes.get("element");
+        if (inputListType != elementType.listOf()) {
+            throw new IllegalArgumentException("The Element's type does not match that of the List.");
+        }
```

Comment:
no need to care about or check for this, is there? contains will be false in this situation.

Reply:
do you think it would be better to enforce this to ensure the user is not using the node incorrectly?

Reply:
if the user is doing something wrong, they should be shown an error before they even execute their flow. in this case, you could make the types change to make connecting it 'wrongly' impossible.

Reply:
I've changed ListContainsNodeType to try to determine the possible input type based on that of the list entered. 
I'm not sure if it would work or not so if you could have a look that would be great.